src/main/java/gov/loc/repository/bagit/reader/BagitTextFileReader.java

```diff
@@ -32,23 +34,63 @@ private BagitTextFileReader(){
    * @throws InvalidBagMetadataException if the bagit.txt file does not conform to the bagit spec
    */
   public static SimpleImmutableEntry<Version, Charset> readBagitTextFile(final Path bagitFile) throws IOException, UnparsableVersionException, InvalidBagMetadataException{
+    final BagitFileValues values = parseValues(bagitFile);
+    
+    return new SimpleImmutableEntry<Version, Charset>(values.version, values.encoding);
+  }
+  
+  /**
+   * Read the bagit.txt file and get the version and encoding. In version 1.0+ also check for
+   * payload-byte-count and payload-file-count
+   * 
+   * @param bag the to read that contains the bagit.txt file and set the values in the bag
+   * 
+   * @throws IOException if there is a problem reading a file
+   * @throws UnparsableVersionException if there is a problem parsing the bagit version number
+   * @throws InvalidBagMetadataException if the bagit.txt file does not conform to the bagit spec
+   */
+  public static void readBagitTextFile(final Bag bag) throws IOException, UnparsableVersionException, InvalidBagMetadataException{
+    Path bagitDir = bag.getRootDir().resolve(".bagit");
+    if(!Files.exists(bagitDir)){
+      bagitDir = bag.getRootDir();
+    }
+    final BagitFileValues values = parseValues(bagitDir.resolve("bagit.txt"));
+    
+    bag.setVersion(values.version);
+    bag.setFileEncoding(values.encoding);
+    bag.setPayloadByteCount(values.payloadByteCount);
+    bag.setPayloadFileCount(values.payloadFileCount);
+  }
+  
+  private static BagitFileValues parseValues(final Path bagitFile) throws UnparsableVersionException, IOException, InvalidBagMetadataException{
     logger.debug("Reading [{}] file", bagitFile);
     final List<SimpleImmutableEntry<String, String>> pairs = KeyValueReader.readKeyValuesFromFile(bagitFile, ":", StandardCharsets.UTF_8);
+    final BagitFileValues values = new BagitFileValues();
     
     String version = "";
     Charset encoding = StandardCharsets.UTF_8;
     for(final SimpleImmutableEntry<String, String> pair : pairs){
       if("BagIt-Version".equals(pair.getKey())){
         version = pair.getValue();
         logger.debug("BagIt-Version is [{}]", version);
+        values.version = parseVersion(version);
       }
       if("Tag-File-Character-Encoding".equals(pair.getKey())){
         encoding = Charset.forName(pair.getValue());
         logger.debug("Tag-File-Character-Encoding is [{}]", encoding);
+        values.encoding = encoding;
+      }
+      if("Payload-Byte-Count".equals(pair.getKey())){ //assume version is 1.0+
+        logger.debug("Payload-Byte-Count is [{}]", pair.getKey());
```

Comment:
(and possibly the whole thing should be something like `"{} is {}", pair.getKey(), pair.getValue()`?